appnav/src/main/kotlin/io/element/android/appnav/RootFlowNode.kt

```diff
@@ -88,11 +90,17 @@ class RootFlowNode @AssistedInject constructor(
     private fun observeLoggedInState() {
         authenticationService.isLoggedIn()
             .distinctUntilChanged()
+            .combine(
+                authenticationService.cacheIdx().onEach {
+                    Timber.v("cacheIdx=$it")
+                    matrixClientsHolder.removeAll()
+                }
+            ) { isLoggedIn, cacheIdx -> isLoggedIn to cacheIdx }
             .onEach { isLoggedIn ->
```

Comment:
This lambda param is now misnamed, it's a Pair, maybe you might want to use the destructuring syntax here.

Reply:
Yes, good catch I will rename it.